src/hotspot/share/c1/c1_LIRGenerator.cpp

```diff
@@ -2279,8 +2098,24 @@ void LIRGenerator::do_UnsafeGetObject(UnsafeGetObject* x) {
   }
 
   LIR_Opr result = rlock_result(x, type);
-  access_load_at(decorators, type,
-                 src, off.result(), result);
+  if (!x->is_raw()) {
+    access_load_at(decorators, type, src, off.result(), result);
+  } else {
+    // Currently it is only used in GraphBuilder::setup_osr_entry_block.
+    // It reads the value from [src + offset] directly.
+#ifdef _LP64
+    LIR_Opr offset = new_register(T_LONG);
+    __ convert(Bytecodes::_i2l, off.result(), offset);
+#else
+    LIR_Opr offset = off.result();
+#endif
+    LIR_Address* addr = new LIR_Address(src.result(), offset, type);
+    if (type == T_LONG || type == T_DOUBLE) {
+      __ unaligned_move(addr, result);
```

Comment:
You're always using unaligned_move. Old code checks x->may_be_unaligned(). May be ok, though.

Reply:
Thank you Martin for running these tests!

In this case, may_be_unaligned is must-be-unaligned since setup_osr_entry_block always creates unaligned UnsafeGetRaw before.